packages/graphql-hooks/src/useClientRequest.js

```diff
@@ -95,64 +96,76 @@ function useClientRequest(query, initialOpts = {}) {
     }
   }, [])
 
+  const latestInitialOpts = React.useRef(null)
+  latestInitialOpts.current = initialOpts
+
+  const latestStateData = React.useRef(null)
+  latestStateData.current = state.data
+
   // arguments to fetchData override the useClientRequest arguments
-  function fetchData(newOpts) {
-    if (!isMounted.current) return Promise.resolve()
-    const revisedOpts = {
-      ...initialOpts,
-      ...newOpts
-    }
+  const fetchData = React.useCallback(
```

Comment:
I think it might be clearer to leave this as-is, but at the end before we return the tuple we do the `useCallback` then:

```js
function fetchData() {...}

const memoisedFetchData = useCallback(fetchData, [client])

return [memoisedFetchData, state]
```

Reply:
Done

Reply:
Although, this refactoring may be dangerous in long term because, as I found out, eslint-plugin-react-hooks will ignore this function. 

```jsx
function fetchData() {...}
const memoisedFetchData = useCallback(fetchData) // it's a mistake, but react-hooks linter rule is silent
```

```jsx
const fetchData = useCallback(() => {}) // it's a mistake and we get react-hooks/exhaustive-deps warning
```